File: crates/breez-sdk/bindings/examples/cli/langs/go/persistence.go
package main

import (
	"fmt"
	"os"
	"path/filepath"

	"github.com/tyler-smith/go-bip39"
)

const (
	phraseFileName  = "phrase"
	historyFileName = "history.txt"
)

// CliPersistence handles mnemonic and history file storage.
type CliPersistence struct {
	dataDir string
}

// GetOrCreateMnemonic reads an existing mnemonic from the data directory,
// or generates a new 12-word BIP39 mnemonic and saves it.
func (p *CliPersistence) GetOrCreateMnemonic() (string, error) {
	filename := filepath.Join(p.dataDir, phraseFileName)

	data, err := os.ReadFile(filename)
	if err == nil {
		return string(data), nil
	}
	if !os.IsNotExist(err) {
		return "", fmt.Errorf("can't read from file %s: %w", filename, err)
	}

	entropy, err := bip39.NewEntropy(128)
	if err != nil {
		return "", fmt.Errorf("failed to generate entropy: %w", err)
	}
	mnemonic, err := bip39.NewMnemonic(entropy)
	if err != nil {
		return "", fmt.Errorf("failed to generate mnemonic: %w", err)
	}

	if err := os.WriteFile(filename, []byte(mnemonic), 0600); err != nil {
		return "", fmt.Errorf("failed to write mnemonic: %w", err)
	}

	return mnemonic, nil
}

// HistoryFile returns the path to the REPL history file.
func (p *CliPersistence) HistoryFile() string {
	return filepath.Join(p.dataDir, historyFileName)
}

File: crates/breez-sdk/bindings/examples/cli/langs/typescript/src/persistence.js
'use strict'

const fs = require('fs')
const path = require('path')
const bip39 = require('bip39')

const PHRASE_FILE_NAME = 'phrase'
const HISTORY_FILE_NAME = 'history.txt'

class CliPersistence {
  /**
   * @param {string} dataDir - Path to the data directory
   */
  constructor(dataDir) {
    this.dataDir = dataDir
  }

  /**
   * Reads an existing mnemonic from the data directory, or generates a new
   * 12-word BIP39 mnemonic and saves it.
   *
   * @returns {string} The mnemonic phrase
   */
  getOrCreateMnemonic() {
    const filename = path.join(this.dataDir, PHRASE_FILE_NAME)

    try {
      const phrase = fs.readFileSync(filename, 'utf-8').trim()
      if (phrase) {
        return phrase
      }
    } catch (err) {
      if (err.code !== 'ENOENT') {
        throw new Error(`Can't read from file ${filename}: ${err.message}`)
      }
    }

    // Generate a new 12-word mnemonic (128 bits of entropy)
    const mnemonic = bip39.generateMnemonic(128)
    fs.mkdirSync(path.dirname(filename), { recursive: true })
    fs.writeFileSync(filename, mnemonic, { mode: 0o600 })
    return mnemonic
  }

  /**
   * Returns the path to the REPL history file.
   *
   * @returns {string} Path to the history file
   */
  historyFile() {
    return path.join(this.dataDir, HISTORY_FILE_NAME)
  }
}

module.exports = { CliPersistence }
